OneSignalSDK/onesignal/src/main/java/com/onesignal/GooglePlayServicesUpgradePrompt.java

```diff
@@ -39,7 +28,7 @@ private static boolean isGooglePlayStoreInstalled() {
    }
 
    static void ShowUpdateGPSDialog() {
-      if (isGMSInstalledAndEnabled() || !isGooglePlayStoreInstalled())
+      if (new OSUtils().getDeviceType() != UserState.DEVICE_TYPE_ANDROID || !isGooglePlayStoreInstalled())
```

Comment:
Maybe we can refactor something along these lines across the codebase?
```
UserState deviceType = new OSUtils().getDeviceType();
if (!deviceType.isAndroidDevice() || !isGooglePlayStoreInstalled())
```

This would make the `getDeviceType()` return `UserState` instead of `int` though
Seems a bit out of scope for this but thought I would bring it up

So maybe...?
```
int deviceType = new OSUtils().getDeviceType();
if (deviceType != UserState.DEVICE_TYPE_ANDROID || !isGooglePlayStoreInstalled())
```

Reply:
Uhm maybe have on OSUtils a method

boolean isAndroidDevice(){
return getDeviceType() == UserState.DEVICE_TYPE_ANDROID;
}

if that make sense

Reply:
@Jeasmine I like your suggestion to clean this up. Using this definition which is `static` and includes `DeviceType` in the name for completeness.

```java
   static boolean isAndroidDeviceType() {
      return new OSUtils().getDeviceType() == UserState.DEVICE_TYPE_ANDROID;
   }
```